### PR TITLE
test(component): Add unit tests for `Sidebar`

### PR DESCRIPTION
**Pull request by tulup-conner**
## Breaking changes:

None.

## Features

- [x] Accept HTML attributes on `Tooltip`s (`...rest`)
- [x] Accept HTML attributes on `Sidebar.ItemGroup`
- [x] Change `Sidebar.Collapse` icon color when open/expanded

## Bug fixes

- [x] Add `public/` folder to Storybook development server, so images are available in stories
- [x] Sidebar
  - [x] Add dark mode for `Sidebar.Item` icons
  - [x] Add accessible name to `Sidebar.Collapse` when `Sidebar` is collapsed
  - [x] Add accessible name to `Sidebar.Logo` when `Sidebar` is collapsed
  - [x] Display `Tooltip` on `Sidebar.Collapse` as well as `Sidebar.Item` when `Sidebar` is collapsed
  - [x] Hide `Sidebar.CTA` when `Sidebar` is collapsed
  - [x] Hide `Sidebar.Logo` text when `Sidebar` is collapsed
  - [x] Remove duplicate Sidebar inside each Sidebar story

### Unit

- [x] Add `Sidebar` unit tests (Resolves #83)

### CODE REVIEW

File: src/lib/components/Sidebar/Sidebar.stories.tsx [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
+);
 
-export const DefaultSidebar = Template.bind({});
-DefaultSidebar.storyName = 'Default';
-DefaultSidebar.args = {
+export const Default = Template.bind({});
+Default.args = {
   children: (
-    <Sidebar collapsed={false}>
+    <SidebarComponent>
```
Comment (rluders) [inline src/lib/components/Sidebar/Sidebar.stories.tsx:23]: I'm very confuse about this `SidebarComponent`... isn't just `Sidebar`? Why?

**Review by rluders — CHANGES_REQUESTED** [review]

File: src/lib/components/Sidebar/Sidebar.stories.tsx (hunk 1 above)
Comment (tulup-conner) [reply src/lib/components/Sidebar/Sidebar.stories.tsx:23]: Yea, agreed.

Comment (tulup-conner) [reply src/lib/components/Sidebar/Sidebar.stories.tsx:23]: The reason this is exported is to accommodate a strange issue with Storybook and React Context. I'm looking for a solution.

**Review by rluders — CHANGES_REQUESTED** [review]
@tulup-conner just let me know when you get done with the arrow functions.

**Comment by tulup-conner** [comment]
> @tulup-conner just let me know when you get done with the arrow functions.

Sorry,I thought that had already gone through. Just pushed that.

File: src/lib/components/Sidebar/index.tsx [hunk 2]
```diff
@@ -1,30 +1,31 @@
...
-    >
-      <div className="h-full overflow-y-auto rounded bg-gray-50 py-4 px-3 dark:bg-gray-800">
-        {childrenAsArray(children).map((child, i) => cloneElement(child as ReactElement, { key: i, collapsed }))}
-      </div>
-    </aside>
+    <SidebarContext.Provider value={{ collapsed }}>
+      <aside aria-label="Sidebar" className={classNames('h-full', collapsed ? 'w-16' : 'w-64')} {...rest}>
+        <div
+          className={classNames(
+            'h-full overflow-y-auto overflow-x-hidden rounded bg-gray-50 py-4 px-3 dark:bg-gray-800',
```
Comment (bacali95) [inline src/lib/components/Sidebar/index.tsx:21]: Please consider using the same sidebar background color as the Flowbite components: white for light mode
https://flowbite.com/application-ui/preview/

**Review by bacali95 — CHANGES_REQUESTED** [review]

File: src/lib/components/Sidebar/index.tsx (hunk 2 above)
Comment (bacali95) [reply src/lib/components/Sidebar/index.tsx:21]: The `bg-gray-50` can be used just for the demo as an override class to differentiate with the example card background color, but the default should be `bg-white`

**Review by bacali95 — APPROVED** [review]

**Review by rluders — APPROVED** [review]

**Pull request merged**